bioimageio_chatbot/chatbot_extensions/hpa_extension.py

```diff
@@ -0,0 +1,112 @@
+                            'Disease involvement', 'Secretome function', 'CCD Protein', 'CCD Transcript',
+                            'Evidence', 'Protein class']
+        
+        info_list = []
+        for index in query_results.index:
+            items = self.data.loc[index, selected_columns]
+            info_list.append(items.to_dict())
+        return info_list
+
+    async def read_protein_info(self,
+        ensembl: str = Field(..., description="Ensembl ID of the protein.")
+    )-> Dict[str, Any]:
+        """Get detailed information about a protein from the Human Protein Atlas."""
+        json_link = f"https://www.proteinatlas.org/{ensembl}.json"
+        response = requests.get(json_link)
+        # check if the request was successful
+        response.raise_for_status()
+        # return the content
+        return response.json()
+
+
+
+    async def get_cell_image(self,
+        gen: str = Field(..., description="Gene name of the protein."),
+        ensembl: str = Field(..., description="Ensembl ID of the protein."),
+        section: str = Field("subcellular", description="Section of the Human Protein Atlas to search for the protein. Valid options are 'subcellular', 'tissue',")
+        ) -> List[str]:
+        """Retrieve a list of cell image links from the Human Protein Atlas, where a specific protein is tagged in the green channel. 
+        The results should be rendered as a horizatal table of images and create link (format: `[![](http://..._thumb.jpg)](http://....jpg)`) to the full-size image without the '_thumb' suffix."""
+        link_name = f"{ensembl}-{gen}"
+        http_link = f"https://www.proteinatlas.org/{link_name}/{section}"
+        # read the source code of the page
+        response = requests.get(http_link)
+        # Search for image links, capturing the part after 'src="'
+        pattern = r'src="(?P<url>//images\.proteinatlas\.org/.*?_red_green_thumb\.jpg)"'
+        image_links = re.findall(pattern, response.text)
+        # replace the 'red_green' with 'blue_red_green_yellow' 
+        image_links = [link.replace('red_green', 'blue_red_green_yellow') for link in image_links]
+        # Remove '_thumb' from each link and print or process them
+        final_image_links = []
+        for link in image_links:
+            final_image_links.append(f"https:{link}")
+        return final_image_links
+
+
+def get_extension():
+    hpa_client = HPAClient()
+    search_tool = schema_tool(hpa_client.search_hpa)
+    read_tool = schema_tool(hpa_client.read_protein_info)
+    get_cell_image_tool = schema_tool(hpa_client.get_cell_image)
+
+    return ChatbotExtension(
+        id="hpa",
+        name="Human Protein Atlas",
+        description="Search the Human Protein Atlas to find detailed protein information, including gene expressions, functions, locations, and disease associations. Get concise, relevant results tailored to your query.",
```

Comment:
I would suggest to change it to `Search the Human Protein Atlas to find human protein-related information, including gene expressions, functions, locations, disease associations, and cell images etc.`